src/Compilers/Core/Portable/Collections/SmallDictionary.cs

```diff
@@ -149,28 +149,18 @@ public AvlNodeHead(int hashCode, K key, V value, Node next)
             public override Node Next => next;
         }
 
-        // separate class to ensure that HashCode field 
-        // is layed out in ram  before other AvlNode fields
-        private abstract class HashedNode : Node
```

Comment:
HashCode was extracted to a intermediate class to make sure it is laid out before the rest of the fields. Presumably to work better with CPU cache since we always read HashCode before other fields.
That, however is a very minor benefit while, it may cost us unnecessary node size increase due to inefficient packing, depending on types of the key and value.

Basically, the JIT will have an opportunity to optimally pack two addresses + int + sbyte, but the hashcode will likely be laid out after the addresses. (addresses should be assumed 64bit and thus 64bit aligned)

If overall effect of the change is negligible, it is a I think the code is a bit simpler too. :-)

Alternatively, the "sbyte Balance" could be moved up to the abstract class.